modules/admin-ui/src/main/java/org/opencastproject/adminui/impl/ThumbnailImpl.java

```diff
@@ -371,38 +394,44 @@ private URI updateOaiPmh(final MediaPackage mp, final MediaPackageElementFlavor
     attachment.setFlavor(flavor);
     tags.forEach(attachment::addTag);
     attachment.setMimeType(tempThumbnailMimeType);
-    if (conversionProfile != null) {
-      downscaleAttachment(conversionProfile, attachment);
+    final Collection<MediaPackageElement> addElements = new ArrayList<>();
+    if (conversionProfiles != null && conversionProfiles.length > 0) {
+      addElements.addAll(downscaleAttachment(attachment, conversionProfiles));
+    } else {
+      addElements.add(attachment);
     }
-
-    final Collection<MediaPackageElement> addElements = Collections.singleton(attachment);
     final Set<String> removeElementsIds = Arrays.stream(pub.getAttachments()).filter(priorFilter)
       .map(MediaPackageElement::getIdentifier).collect(Collectors.toSet());
     final Publication newPublication = this.configurablePublicationService.replaceSync(mp, channelId, addElements, removeElementsIds);
     mp.remove(pub);
     mp.add(newPublication);
     //noinspection ConstantConditions
-    final Attachment newElement = Arrays.stream(newPublication.getAttachments())
-      .filter(att -> att.getIdentifier().equals(aid)).findAny().get();
-    return Tuple.tuple(aUri, newElement);
+    final Set<String> newAttachmentIds = addElements.stream()
+      .map(MediaPackageElement::getIdentifier)
+      .collect(Collectors.toSet());
+    return Tuple.tuple(aUri, Arrays.stream(newPublication.getAttachments())
+      .filter(att -> newAttachmentIds.contains(att.getIdentifier()))
+      .collect(Collectors.toList()));
   }
 
-  private void downscaleAttachment(final String conversionProfile, final Attachment attachment)
-    throws EncoderException, MediaPackageException {
+  private List<Attachment> downscaleAttachment(final Attachment attachment, final String... conversionProfiles)
+    throws DistributionException, EncoderException, MediaPackageException {
     // What the composer returns is not our original attachment, modified, but a new one, basically containing just
     // a URI.
-    final Attachment downscaled = composerService.convertImageSync(attachment, conversionProfile);
-    attachment.setURI(downscaled.getURI());
+    final List<Attachment> downscaled = composerService.convertImageSync(attachment, conversionProfiles);
+    return downscaled.stream().map(a -> cloneAttachment(attachment, a.getURI())).collect(Collectors.toList());
   }
 
   private URI updateExternalPublication(final MediaPackage mp, final MediaPackageElementFlavor trackFlavor)
```

Comment:
Maybe rename this to updateConfigurablePublication?

Reply:
Fixed.